docs/guides/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.md

```diff
@@ -0,0 +1,445 @@
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'Python', value:'python'},
+  ]}>
+<TabItem value="yaml">
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.py#L77-L94
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.py#L105-L123
+```
+
+</TabItem>
+</Tabs>
+
+Running `test_yaml_config()` would result in 1 Data Asset `yellow_tripdata` with 3 associated data_references: `yellow_tripdata_2019-01.csv`, `yellow_tripdata_2019-02.csv` and `yellow_tripdata_2019-03.csv`, seen also in Example 1 below.
+
+A corresponding configuration for `InferredAssetS3DataConnector` would look similar but would require `bucket` and `prefix` values instead of `base_directory`.
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'Python', value:'python'},
+  ]}>
+<TabItem value="yaml">
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.py#L147-L165
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.py#L178-L197
+```
+
+</TabItem>
+</Tabs>
+
+The following examples will show scenarios that InferredAssetDataConnectors can help you analyze, using `InferredAssetFilesystemDataConnector`.
+
+
+### Example 1: Basic configuration for a single Data Asset
+
+Continuing the example above, imagine you have the following files in the directory `<MY DIRECTORY>`:
+
+```
+<MY DIRECTORY>/yellow_tripdata_2019-01.csv
+<MY DIRECTORY>/yellow_tripdata_2019-02.csv
+<MY DIRECTORY>/yellow_tripdata_2019-03.csv
+```
+
+Then this configuration...
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'Python', value:'python'},
+  ]}>
+<TabItem value="yaml">
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.py#L225-L242
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.py#L253-L271
+```
+
+</TabItem>
+</Tabs>
+
+...will make available `yelow_tripdata` as a single Data Asset with the following data_references:
+
+```bash
+Available data_asset_names (1 of 1):
+    yellow_tripdata (3 of 3): ['yellow_tripdata_2019-01.csv', 'yellow_tripdata_2019-02.csv', 'yellow_tripdata_2019-03.csv']
+
+Unmatched data_references (0 of 0):[]
+```
+
+Once configured, you can get `Validators` from the `Data Context` as follows:
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_an_inferredassetdataconnector.py#L294-L303
+```
```

Comment:
This is now addressed in lines 268-288

Reply:
Awesome! Do you think it's worth including an example for specifying based on the full `csv` name as well? (Apologies if you have this somewhere and I didn't see it).

Reply:
The only way I can find to do this is something like this:

```
from great_expectations.datasource.data_connector.util import (
    convert_data_reference_string_to_batch_identifiers_using_regex,
)

batch_request = BatchRequest(
    datasource_name="taxi_datasource",
    data_connector_name="default_inferred_data_connector_name",
    data_asset_name="yellow_tripdata",
    data_connector_query={
        "custom_filter_function": convert_data_reference_string_to_batch_identifiers_using_regex(
            data_reference="yellow_tripdata_2019-02.csv",
            regex_pattern="(.*)_(\d{4})-(\d{2})\.csv",
            group_names=["data_asset_name", "year", "month"],
        )
    },
)
```

is that what you are referring to?

Reply:
Yikes! No, I thought this might be possible if your DataConnector is set up like this:

```yaml
default_inferred_data_connector_name:
        module_name: great_expectations.datasource.data_connector
        base_directory: ../my_data_dir
        default_regex:
          group_names:
            - data_asset_name
          pattern: (.*)
        class_name: InferredAssetFilesystemDataConnector
```

and then in your BatchRequest, your group_name is just the entire name of the file. If this is too much for this example, feel free to disregard.

Reply:
I think what we have shown should be sufficient to explain that the group_names are used in the batch_identifiers and if you only have one group, you can pass in the entire file name